site/en/blog/deprecating-web-sql/index.md

```diff
@@ -0,0 +1,238 @@
+  is planned for later Chromium releases.
+authors:
+  - thomassteiner
+date: 2022-08-26
+# updated: 2022-08-26
+hero: image/8WbTDNrhLsU0El80frMBGE4eMCD3/yUp8lfaCt4EmxmVei3lj.jpg
+alt: Filing cabinet symbolizing a database.
+tags:
+  - deprecations
+  - removals
+  - storage
+---
+
+{% Aside %} This is a living post that will be updated as the deprecation steps outlined below
+happen. {% endAside %}
+
+The [Web SQL Database API](https://www.w3.org/TR/webdatabase/), which allows you to store data in a
+structured manner on the user's computer (internally based on the SQLite database engine), was
+[introduced in April 2009](https://www.w3.org/TR/2009/WD-webdatabase-20091222/) and
+[abandoned in November 2010](https://www.w3.org/TR/webdatabase/#status-of-this-document). While it
+was implemented in WebKit (which powers Safari and early versions of Chrome) and remained active in
+the Blink engine (that powers Chrome after the switch from WebKit), Gecko (which powers Firefox)
+never implemented this feature and
+[WebKit removed it in 2019](https://lists.webkit.org/pipermail/webkit-dev/2019-November/030968.html).
+
+The World Wide Web Consortium (W3C)
+[encourages](https://www.w3.org/TR/webdatabase/#:~:text=The%20Web%20Applications%20Working%20Group%20continues%20work%20on%20two%20other%20storage%2Drelated%20specifications%3A%20Web%20Storage%20and%20Indexed%20Database%20API.)
+those needing web databases to adopt
+[Web Storage API](https://developer.mozilla.org/docs/Web/API/Web_Storage_API) technologies like
+[`localStorage`](https://developer.mozilla.org/docs/Web/API/Window/localStorage) and
+[`sessionStorage`](https://developer.mozilla.org/docs/Web/API/Window/sessionStorage), or
+[IndexedDB](https://developer.mozilla.org/docs/Web/API/IndexedDB_API/Using_IndexedDB). These
+technologies show their strenghts when it comes to key/value stores and structured data, but
+acknowledgely also have weaknesses like the lack of a strong query language. People want SQL on the
+web for a reason.
+
+{% Aside %} Our intention is to empower developers to create their own solutions for structured
+storage and we're therefore working with the [SQLite](https://www.sqlite.org/index.html) team to
+create a SQLite implementation over WebAssembly. This solution will replace Web SQL. {% endAside %}
+
+## Web SQL deprecation steps
+
+- [✅ Done.] Web SQL was deprecated and removed for **third-party contexts** in
+  **Chromium&nbsp;97**.
+- [📍 We are here.] Web SQL access in **insecure contexts** is deprecated as of
+  **Chromium&nbsp;105** at which time a warning message will be shown in the Chrome DevTools Issue
+  panel.
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/unDM9iaTcPDbedQ3dL84.png", alt="Chrome DevTools Issues panel with a warning that reads Web SQL in non-secure contexts is deprecated and will be removed in M107. Please use Web Storage or Indexed Database.", width="800", height="158" %}
```

Comment:
If the `alt` text matches the image, we need to remove this, since it mentions the outdated M107 milestone. Sadly I just discovered that this warning is actually appearing in M105 beta, but we're going to change it in the next couple of weeks.

Sorry this has been so complicated!

Reply:
In this case let's just shorten the `alt` text.